dask_jobqueue/lsf.py

```diff
@@ -0,0 +1,111 @@
+    job_mem : str
+        Request memory. Passed to `#BSUB -M` option.
+    job_extra : list
+        List of other LSF options, for example -e. Each option witll be
+        prepended with the #LSF prefix.
+    %(JobQueueCluster.parameters)s
+
+    Examples
+    --------
+    >>> from dask_jobqueue import LSFCluster
+    >>> cluster = LSFcluster(queue='general', project='DaskonLSF')
+    >>> cluster.start_workers(10)  # this may take a few seconds to launch
+
+    >>> from dask.distributed import Client
+    >>> client = Client(cluster)
+
+    This also works with adaptive clusters.  This automatically launches and
+    kill workers based on load.
+
+    >>> cluster.adapt()
+    """, 4)
+
+    # Override class variables
+    submit_command = 'bsub'
+    cancel_command = 'bkill'
+    scheduler_name = 'lsf'
+
+    def __init__(self, queue=None, project=None, walltime=None,
+                 job_cpu=None, job_mem=None, job_extra=None, **kwargs):
+        if queue is None:
+            queue = dask.config.get('jobqueue.lsf.queue')
+        if project is None:
+            project = dask.config.get('jobqueue.lsf.project')
+        if walltime is None:
+            walltime = dask.config.get('jobqueue.lsf.walltime')
+        if job_cpu is None:
+            job_cpu = dask.config.get('jobqueue.lsf.job-cpu')
+        if job_mem is None:
+            job_mem = dask.config.get('jobqueue.lsf.job-mem')
+        if job_extra is None:
+            job_extra = dask.config.get('jobqueue.lsf.job-extra')
+
+        # Instantiate args and parameters from parent abstract class
+        super(LSFCluster, self).__init__(**kwargs)
+
+        # LSF header build
+        header_lines = []
+        if self.name is not None:
+            header_lines.append('#BSUB -J %s' % self.name)
+            header_lines.append('#BSUB -e %s.err' % self.name)
+            header_lines.append('#BSUB -o %s.out' % self.name)
+        if queue is not None:
+            header_lines.append('#BSUB -q %s' % queue)
+        if project is not None:
+            header_lines.append('#BSUB -P %s' % project)
+        if walltime is not None:
+            header_lines.append('#BSUB -W %s' % walltime)
+        ncpus = job_cpu
+        if ncpus is None:
+            ncpus = self.worker_processes * self.worker_threads        
+        header_lines.append('#BSUB -n %s' % ncpus)
+        memory_string = job_mem
+        if memory_string is None and self.worker_memory is not None:
+            memory = self.worker_processes * self.worker_memory
+            memory_string = lsf_format_bytes_ceil(memory)
+        header_lines.append('#BSUB -M %s' % memory_string)
+        header_lines.extend(['#BSUB %s' % arg for arg in job_extra])
+
+        # Declare class attribute that shall be overriden
+        self.job_header = '\n'.join(header_lines)
+
+        logger.debug("Job script: \n %s" % self.job_script())
+
+    def _job_id_from_submit_output(self, out):
+        return out.split('.')[0].strip()
```

Comment:
Here you need to either do a parsing of the output string of the job, which seems to be something like:
````
Job is submitted to <cpp> project.
Job <16633929> is submitted to default queue <general>.
````
But you more likely want to find the correct option for just outputing the job ID after the bsub command. Unfortunatly I did not find such an option after a few minutes Google search.